### DISCUSSION TITLE
Can we create index for external parquet files?

### DISCUSSION THREAD

**Discussion by kesavkolla** [duckdb/duckdb#4820] [Q&A]
I have lots of parquet files in s3. Is it possible to create index on some fields and store the index in duckdb? I want duckdb to use that index while querying data from parquet. Is it something duckdb can support?

**Comment by hannes** [comment]
This is not supported yet, but we are thinking about it

**Reply by kesavkolla** [reply]
This feature would be really helpful a lot for larger datasets.

**Comment by SteveLauC** [comment]
Hi, I am also thinking about this, this does not seem to be something common (or possibly practical) as a Google search didn't show me too many results. Take the most common B+Tree index as an example, what should be stored in the index entry as Parquet is not row-based, it is more like a hybrid structure, maybe a row group number could be used?

Is there still a plan for DuckDB to support this? And, I would like to hear your thoughts on this feature:)

**Comment by davidberenstein1957** [comment]
I am having the same issue. Wanting to host vector search indices locally, while having remote data on Hugging Face. @hannes is this something that still will potentially be implemented?

**Comment by turnerlewis** [comment]
Would be really nice
